src/sentry/api/endpoints/organization_member/details.py

```diff
@@ -248,10 +261,16 @@ def put(
                     status=403,
                 )
 
-            if member.user == request.user and (result["role"] != member.role):
+            if member.user == request.user and (assigned_role != member.role):
                 return Response({"detail": "You cannot make changes to your own role."}, status=400)
 
-            self._change_org_member_role(member, result["role"])
+            if organization_roles.get(assigned_role).is_retired and features.has(
+                "organizations:team-roles", organization
+            ):
```

Comment:
If the role is the role the user already has, should we just no-op here instead of erroring?